closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java

```diff
@@ -99,7 +101,7 @@
     }
 
     private static void checkDataLength(int processed, int len) {
-        if (processed > MAX_BUF_SIZE - len) {
+        if (processed + len > MAX_BUF_SIZE) {
```

Comment:
This is because `MAX_BUF_SIZE` is set to `Integer.MAX_VALUE`. Hence, when `len` is negative, `MAX_BUG_SIZE + len` overflows, which triggers the if statement and throws an exception. Hence, I thought it would make sense to move the `len` calculation to the other side.

A side note here: `len` can be negative because of [this line](https://github.com/ibmruntimes/openj9-openjdk-jdk/blob/d328410ecd9a3ea77cdd67f97ea69d96842ac6dc/closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java#L487). When `len` (in the referenced line) is zero, the second argument passed to `checkDataLength` is negative.

However, looking at your comment, it makes sense that `processed + len` can also overflow. In this case, I think it's best to add a check for negative values to handle this case. So that would become:

```java
if(len < 0) return;
if (processed > MAX_BUF_SIZE - len) throw new ProviderException("...");
```

Reply:
@pshipton Please let me know if you have any concerns about this new implementation (I updated the branch with the fix).